components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1697,10 +1729,28 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
     @Override
     public List<Object> listGroupsWithGET(Node rootNode, int startIndex, int count, String sortBy, String sortOrder,
             String domainName, Map<String, Boolean> requiredAttributes)
+
             throws CharonException, NotImplementedException, BadRequestException {
         if (sortBy != null || sortOrder != null) {
             throw new NotImplementedException("Sorting is not supported");
-        } else if (startIndex != 1) {
+        } else if (startIndex != 1 && count >= 0) {
+            throw new NotImplementedException("Pagination is not supported");
+        } else if (rootNode != null) {
+            return filterGroups(rootNode, requiredAttributes);
+        } else {
+            return listGroups(requiredAttributes);
+        }
+    }
+
+    @Override
+    public List<Object> listGroupsWithGET(Node rootNode, Integer startIndex, Integer count, String sortBy,
+            String sortOrder, String domainName, Map<String, Boolean> requiredAttributes)
+            throws CharonException, NotImplementedException, BadRequestException {
+
+
+        if (sortBy != null || sortOrder != null) {
+            throw new NotImplementedException("Sorting is not supported");
+        } else if (startIndex != 1  || count != null) {
```

Comment:
As both startIndex and count are instances of Integer it would be better to check for not null of each

Reply:
fixed: 8f8c20292c4fac76778a8aa95490675023d7d063